src/Serilog/Sinks/PeriodicBatching/PeriodicBatchingSink.cs

```diff
@@ -216,11 +213,8 @@ void OnTick()
                     while (_queue.TryDequeue(out evt)) { }
                 }
 
-                lock (_stateLock)
-                {
-                    if (!_unloading)
-                        SetTimer(_status.NextInterval);
-                }
+                if (_state == Started)
```

Comment:
I think this is racy - the thread responsible for closing/unloading may have switched state to Unloading between the volatile read and the SetTimer call on the next line, meaning we may end up trying to restart the timer after the unload sequence has run.